src/psyclone/domain/lfric/algorithm/psyir.py

```diff
@@ -0,0 +1,115 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford STFC Daresbury Lab
+
+'''This module contains LFRic Algorithm-layer-specific PSyIR classes.
+
+'''
+from psyclone.domain.common.algorithm import (AlgorithmInvokeCall,
+                                              KernelLayerRef)
+
+
+class LfricAlgorithmInvokeCall(AlgorithmInvokeCall):
+    '''An invoke call in an LFRic Algorithm layer.
+
+    :param routine: the routine that this call calls.
+    :type routine: py:class:`psyclone.psyir.symbols.RoutineSymbol`
+    :param parent: parent of this node in the PSyIR.
+    :type parent: sub-class of :py:class:`psyclone.psyir.nodes.Node`
+    :param str description: xxx
+
+    '''
+    _text_name = "LfricAlgorithmInvokeCall"
+
+    def __init__(self, routine, parent=None, description=None):
+        self._description = description
+        super(LfricAlgorithmInvokeCall, self).__init__(routine, parent=parent)
+
+    @classmethod
+    def create(cls, routine, arguments, description=None):
+        '''Create an instance of class cls given valid instances of a routine
+        symbol, a list of child nodes for its arguments and an
+        optional name.
+
+        :param routine: the routine that class cls calls.
+        :type routine: py:class:`psyclone.psyir.symbols.RoutineSymbol`
+        :param arguments: the arguments to this routine. These are \
+            added as child nodes.
+        :type arguments: list of :py:class:`psyclone.psyir.nodes.DataNode`
+        :param description: a string describing the purpose of the \
+            invoke or None if one is not provided. This is used to \
+            create the name of the subroutine that replaces the \
+            invoke. Defaults to None.
+        :type name: str or NoneType
+
+        :returns: an instance of cls.
+        :rtype: \
+            :py:class:`psyclone.psyir.nodes.LfricAlgorithmInvokeCall` or a \
+            subclass thereof.
+
+        '''
+        instance = super(LfricAlgorithmInvokeCall, cls).create(
+            routine, arguments)
+        instance._description = description
```

Comment:
If there is not going to be a description setter, you can disable the pylint warning.

Reply:
Good idea.